content/4_Gold/DP_Bitmasks.mdx

```diff
@@ -124,6 +124,54 @@ int main() {
 ```
 
 </CPPSection>
+<PySection>
+
+<Warning>
+
+Due to Python's large constant factor, the solution below TLEs on a couple of test cases.
+
+</Warning>
+
+```py
+import sys
+
+input = sys.stdin.readline
+
+MOD = 10**9 + 7
+
+n, m = map(int, input().strip().split())
+dp = [[0 for _ in range(n)] for _ in range(1 << n)]
+dp[1][0] = 1
+
+adj = [[] for _ in range(n)]
+
+for _ in range(m):
+	a, b = map(int, input().strip().split())
+	adj[b - 1].append(a - 1)
+
+for cities in range(1, 1 << n):
+	# only consider cities that include the first city
+	if not (cities & 1):
+		continue
+
+	# only consider subsets with the last city if it's the full subset
+	if cities & (1 << n - 1) and cities != (1 << n) - 1:
+		continue
+
+	# loop through possible ending cities
+	for end in range(1, n):
+		if cities & (1 << end):
```

Comment:
could you do the same condition invertion here

Reply:
Got it, I will also do the same on another similar line below that too